client/allocrunner/taskrunner/service_hook.go

```diff
@@ -140,7 +140,7 @@ func (h *serviceHook) Update(ctx context.Context, req *interfaces.TaskUpdateRequ
 	return h.consul.UpdateTask(oldTaskServices, newTaskServices)
 }
 
-func (h *serviceHook) Killing(ctx context.Context, req *interfaces.TaskPreKillRequest, resp *interfaces.TaskPreKillResponse) error {
+func (h *serviceHook) PreKilling(ctx context.Context, req *interfaces.TaskPreKillRequest, resp *interfaces.TaskPreKillResponse) error {
```

Comment:
What's the cause of this change?  Was it a missed regression?

In the future, would be nice to have a better way to detect hook interface mismatch with type system or other means.

Reply:
The method on the interface was changed in #5207 and this implementation was missed. We do static assertions in tests, but this hook hadn't implemented those yet: https://github.com/hashicorp/nomad/pull/5319/files#diff-5cc9be3181b1e0e392ff275b17fbdfe1R12

Any other ideas on how to avoid mistakes like this are welcome, but we should always have some sort of integration test (like the one I added that discovered this bug) to catch it.

Reply:
good point.  Static checks are quite useful as a cheap baseline.  Would be useful to do a quick audit of all hooks as well.  Thanks for catching it!